xlators/mgmt/glusterd/src/glusterd-handshake.c

```diff
@@ -1948,7 +1948,7 @@ gd_validate_peer_op_version(xlator_t *this, glusterd_peerinfo_t *peerinfo,
     if (!this) {
         gf_smsg("glusterd", GF_LOG_ERROR, errno, GD_MSG_XLATOR_NOT_DEFINED,
                 NULL);
-        goto out;
+        return ret;
```

Comment:
Simply using `GF_ASSERT(this)` at line:1941 would be a better fix, I think.

Reply:
+1

Reply:
I don't think so. Our `GF_ASSERT()` implementation for production builds is simply an `if` and a log message if the condition is false, but the execution flow continues. From the point of view of coverity, this has the same problem: we would have checked for NULL and then dereferenced it.

Either we assume that `this` is never NULL (as it should be) and we don't check it, or we correctly handle the case. We could also use `GF_ABORT()` when it's NULL, which causes a core dump and stops the program, but this will already happen when it's first dereferenced.

Reply:
Actually this function is invoked by __glusterd_mgmt_hndsk_version_cbk wherein this is derived from THIS. So I think GF_ASSERT isn't required.

Reply:
...and it would have crashed here - https://github.com/gluster/glusterfs/blob/3813ffafb12d20d2b728d43b3f0eb5201e6fb7c2/xlators/mgmt/glusterd/src/glusterd-handshake.c#L2125 otherwise... Surely can be removed.

Reply:
At other places too, we are using `GF_ASSERT` to check for NULL and it does crash if it's NULL.
Though, that's a different aspect that we might never get `this` as NULL here.